core/orchestrator.go

```diff
@@ -264,23 +264,25 @@ func (orch *orchestrator) PriceInfo(sender ethcommon.Address) (*net.PriceInfo, e
 	}
 
 	if monitor.Enabled {
-		monitor.TranscodingPrice(sender.String(), price)
+		monitor.TranscodingPrice(sender.String(), common.FixedToPrice(price))
 	}
 
 	return &net.PriceInfo{
-		PricePerUnit:  price.Num().Int64(),
-		PixelsPerUnit: price.Denom().Int64(),
+		PricePerUnit:  price,
+		PixelsPerUnit: common.PriceScalingFactor,
 	}, nil
 }
 
-func (orch *orchestrator) priceInfo(sender ethcommon.Address) (*big.Rat, error) {
+// priceInfo returns price per pixel as a fixed number
+func (orch *orchestrator) priceInfo(sender ethcommon.Address) (int64, error) {
 	txCostMultiplier, err := orch.node.Recipient.TxCostMultiplier(sender)
 	if err != nil {
-		return nil, err
+		return 0, err
 	}
 	// pricePerPixel = basePrice * (1 + 1/ txCostMultiplier)
 	overhead := new(big.Rat).Add(big.NewRat(1, 1), new(big.Rat).Inv(txCostMultiplier))
-	return new(big.Rat).Mul(orch.node.GetBasePrice(), overhead), nil
+	price := new(big.Rat).Mul(orch.node.GetBasePrice(), overhead)
+	return common.PriceToFixed(price)
```

Comment:
Since we're calling `common.FixedToPrice()` on the value returned by this method in both `TicketParams()` and `PriceInfo()`, it might make sense to move the `common.FixedToPrice()` call into this method.

Ex.

```golang
...
fixedPrice, err := common.PriceToFixed(new(big.Rat).Mul(orch.node.GetBasePrice(), overhead))
if err != nil {
    return nil, err
}

return common.FixedToPrice(fixedPrice), nil
```

If we do this I don't think additional changes would be necessary in the callers of `priceInfo()`.

Reply:
This is quite a lot of conversions but I guess I'd be fine with this because it does reduce the changes in the callers. Does this reconversion to a big.Rat uphold the guarantee that we end up with a fraction with a denominator of `common.PriceScalingFactor` or smaller (big.Rat will always seek out the smallest possible denominator) ?

Reply:
fixed in 55ec1c6 and unexported `common.PriceScalingFactor` in 141c824

Reply:
> This is quite a lot of conversions but I guess I'd be fine with this because it does reduce the changes in the callers.

Given that previously we would've had to call `common.FixedToPrice()` in the callers since certain APIs require the price to be passed as `big.Rat` I actually think doing the conversion in `priceInfo()` leads to less conversions overall since we only need to do it once in `priceInfo()`.

> Does this reconversion to a big.Rat uphold the guarantee that we end up with a fraction with a denominator of common.PriceScalingFactor or smaller and a numerator of fixedPrice or smaller (big.Rat will always seek out the smallest possible denominator) ?

I think so. Even if the internal representation in a `big.Rat` uses the lowest denominator, the numerator will never be greater than `fixedPrice` and the denominator will never be greater than the price scaling factor.